File: src/com/colacelli/ircbot/plugins/autoresponse/AutoResponsePluginHelp.java
package com.colacelli.ircbot.plugins.autoresponse;

import com.colacelli.ircbot.plugins.help.PluginHelp;

public class AutoResponsePluginHelp extends PluginHelp {
    static final String SEPARATOR = "|";

    public AutoResponsePluginHelp(String command, int access, String help, String... args) {
        super(command, access, help, args);
    }

    @Override
    public String toString() {
        StringBuilder text = new StringBuilder();
        text.append(command);
        text.append(" ");

        for (int i = 0; i < args.length; i++) {
            String arg = args[i];

            if (i != 0) text.append(SEPARATOR);

            text.append("<");
            text.append(arg);
            text.append(">");
        }

        text.append(": ");
        text.append(help);

        return text.toString();
    }
}

File: src/com/colacelli/ircbot/plugins/autoresponse/AutoResponsePluginHelp.kt
package com.colacelli.ircbot.plugins.autoresponse

import com.colacelli.ircbot.plugins.access.IRCBotAccess
import com.colacelli.ircbot.plugins.help.PluginHelp

class AutoResponsePluginHelp(command : String, access : IRCBotAccess.Level, help : String, vararg args : String) : PluginHelp(command, access, help, *args) {
    companion object {
        const val SEPARATOR = "|"
    }

    override fun toString() : String {
        var text = "$command "

        for (i in 0 until args.size) {
            val it = args[i]

            if (i > 0) text += SEPARATOR

            text += "<$it>"
        }

        text += ": $help"

        return text
    }
}